pkg/sql/parser/sql.y

```diff
@@ -5055,6 +5057,26 @@ drop_type_stmt:
   }
 | DROP TYPE error // SHOW HELP: DROP TYPE
 
+// %Help: DROP TENANT - remove a tenant
+// %Category: DDL
+// %Text: DROP TENANT [IF EXISTS] <name>
+drop_tenant_stmt:
+  DROP TENANT name
+  {
+    $$.val = &tree.DropTenant{
```

Comment:
Referential nit: Same comment here about hiding this.  Should we add a `SKIP DOC` here?

Reply:
I was under the impression from https://github.com/cockroachdb/cockroach/pull/91595#issuecomment-1311940233 that this wouldn't automatically get pushed to the docs.  But I wasn't aware of SKIP DOC. I'll take a look at that.

Reply:
I think we're safe from a docs point of view, but I also just went ahead and added SKIP DOC anyways - https://github.com/cockroachdb/cockroach/pull/91995

Reply:
@ajstorm does it make sense to not SKIP DOC here, so that the UX would feel more complete? it would be nice to have an interface that supports things like `\h DROP TENANT`.

Reply:
@lidorcarmel my thinking was that if the command isn't in the docs, we should hide it entirely (including from the help) as it could be confusing for users who trip over the help.  That being said, I'm not sure what the precedent is here.

Reply:
I see, good point.
FWIW I think we decided to leave the help in for now and remove it later if needed.